### DISCUSSION TITLE
Differences between esm.sh and self hosting

### DISCUSSION THREAD

**Discussion by oravecz** [esm-dev/esm.sh#940] [General]
I am hosting esm.sh internal in our company. Building Fluent UI from Microsoft works on the public esm.sh server, but not on my internal version.

```shell
curl 'http://localhost:5050/@fluentui/react-shared-contexts@9.20.2'
"../../.esmd/npm/@fluentui/react-shared-contexts@9.20.2/node_modules/.pnpm/@fluentui+react-shared-contexts@9.20.2_@types+react@18.3.12_react@18.3.1/node_modules/@fluentui/react-shared-contexts/lib/ThemeContext.js" for import "ThemeContext"
```

I am running the server from command line with 
* Node 22.10.0
* go version go1.22.5 darwin/arm64
* esm.sh v135_7

ESM Config
```json
{
  "port": 5050
}
```

**Comment by ije** [comment]
<img width="1145" alt="Screenshot 2024-12-02 at 22 37 19" src="https://github.com/user-attachments/assets/12ddafc4-f6b8-4bf8-ad4d-339efad6f0a5">
i also can't build it on my computer(macos m1), will look in to it!

**Comment by oravecz** [comment]
Looking as best I could, I am unsure how the platform plays a role in being able to resolve the `import "ThemeContext"`.


The `react-shared-contexts/index.js` is the default export for the package, and it contains the following export.

`export { ThemeContext as ThemeContext_unstable, ThemeProvider as ThemeProvider_unstable } from './ThemeContext';`

So, I am unsure what (in this package) is attempting to resolve `import "ThemeContext"

Sadly, the latest version (9.21.1) has a different issue.

```sh
curl 'http://localhost:5050/@fluentui/react-shared-contexts@9.21.1'
esbuild: Detected cycle while resolving import "CustomStyleHooksContext"
```
